homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,50 @@
+#include <iostream>
+#include "compression.hpp"
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& image) {
+	std::vector<std::pair<uint8_t, uint8_t>> compressedImage;
+	std::pair<uint8_t, uint8_t> tempPair{image.front().front(), 0};
+
+	for (const auto& verticalElement : image) {
+		for (const auto& horizontalElement : verticalElement) {
+			if (horizontalElement == tempPair.first) {
+				++tempPair.second;
+			} else { 
+				compressedImage.push_back(tempPair);
+				tempPair.first = horizontalElement;
+				tempPair.second = 1;
+			}
+		}
+		compressedImage.push_back(tempPair);
+		tempPair.first = verticalElement.front();
+		tempPair.second = 0;
```

Comment:
This is almost same part of code as above, please refactor it and use in one if-else statement